logfire/_internal/tracer.py

```diff
@@ -171,6 +171,12 @@ def record_exception(
         timestamp = timestamp or self.ns_timestamp_generator()
         record_exception(self.span, exception, attributes=attributes, timestamp=timestamp, escaped=escaped)
 
+    def __exit__(self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: Any) -> None:
+        if self.is_recording() and isinstance(exc_value, BaseException):
+            self.record_exception(exc_value, escaped=True)
+
+        self.end()
```

Comment:
```suggestion
        if self.is_recording():
            if isinstance(exc_value, BaseException):
                self.record_exception(exc_value, escaped=True)
            self.end()
```
I confirmed that this would get us back to the old timestamp behavior, again, not sure if this is better or worse than the version currently in this PR. But I think one of these two options should be okay

Reply:
I decided to make this change because, 1, it preserves the old behavior, and 2, I think it marginally reduces overhead. We can always just dedent the call to `self.end()` if we think the method should be called even if the inner span is not a recording span.